spec/presenters/tree_builder_ae_class_spec.rb

```diff
@@ -22,4 +22,25 @@
       domains.should match_array %w(LUIGI MARIO)
     end
   end
+
+  context "#x_get_tree_roots" do
+    before do
+      root_tenant = EvmSpecHelper.create_root_tenant
+      user = FactoryGirl.create(:user)
+      User.stub(:current_user).and_return(user)
+      user.stub(:current_tenant).and_return(root_tenant)
+      tenant2 = FactoryGirl.create(:tenant, :parent => root_tenant)
+      FactoryGirl.create(:miq_ae_domain, :name => "test1", :tenant => root_tenant)
+      FactoryGirl.create(:miq_ae_domain, :name => "test2", :tenant => root_tenant)
+      FactoryGirl.create(:miq_ae_domain, :name => "test3", :tenant => tenant2)
+      FactoryGirl.create(:miq_ae_domain, :name => "test4", :tenant => tenant2)
+    end
+
+    it "should only return domains in a user's current tenant" do
+      tree = TreeBuilderAeClass.new("ae_tree", "ae", {})
+      domains = JSON.parse(tree.tree_nodes).first['children'].collect { |h| h['title'] }
+      domains.should match_array %w(test1 test2)
+      domains.should_not include %w(test3 test4)
```

Comment:
Is there any difference between test1 and test2 or test3 and test4? Let's create one of each type unless there's a difference.